Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+            domain, args = args[0], args[1:]
+            if not args:
+                return cls.from_domain(domain, **kwargs)
+            if isinstance(args[0], TableBase):
+                return cls.from_table(domain, *args)
+            elif isinstance(args[0], list):
+                return cls.from_list(domain, *args)
+            elif isinstance(args[0], pd.DataFrame):
+                return cls.from_dataframe(domain, args[0], **kwargs)
+        else:
+            domain = None
+        return cls.from_numpy(domain, *args, **kwargs)
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the object.
+
+        Sets the name and other attributes and calls pandas' initialization.
+        Filters the domain: retains only the columns actually in the table.
+        Sets the weights if they are not set already.
+        """
+        # see the comment in __new__ for the rationale here
+        # also, another tidbit is that pandas has some internals that need to be set up
+        # and expects its arguments to be set appropriately
+        # because we override the constructor arguments in a completely incompatible way,
+        # we need to pass ourselves as the data object if we have already set things up
+        # previously by e.g. creating an empty Table via the __new__ hack in from_X
+        # functions, then filling up with columns.
+        # to check for this, we check for domain existence because tables without domains
+        # can't really be used in Orange in any meaningful way
+        if hasattr(self, 'domain'):
+            kwargs['data'] = self
+
+        # only pass through things known to pandas, e.g.
+        # Table(..., weights=1) passes weights to from_numpy, but would error
+        # when passing to pandas upstream
+        super().__init__(**{k: v for k, v in kwargs.items() if k in self._KNOWN_PANDAS_KWARGS})
+
+        self.name = getattr(self, 'name', kwargs.get("name", "untitled"))
+        self.attributes = getattr(self, 'attributes', kwargs.get("attributes", {}))
+        self.__file__ = getattr(self, '__file__', kwargs.get("__file__"))
+
+        # we need to filter the domain to only include the columns present in the table
+        # but we still need to allow constructing an empty table (with no domain)
+        # also, we only set the domain if it has changed (==number of variables),
+        # so in those cases, id(domain_before) == id(domain_after)
+        if hasattr(self, 'domain') and self.domain is not None:
+            new_domain = Domain(
+                [c for c in self.domain.attributes if c in self.columns],
+                [c for c in self.domain.class_vars if c in self.columns],
+                [c for c in self.domain.metas if c in self.columns]
+            )
+            if len(new_domain.variables) + len(new_domain.metas) != \
+                            len(self.domain.variables) + len(self.domain.metas):
+                self.domain = new_domain
```

Comment:
So we can strip some data by calling `Table(subset_domain, orig_data)`. Or we can strip the domain by `Table(whatever_domain_might_be_reduced, subset_data)`? Do we rely on this somewhere? Did't it add a new data column of nans in this case before?